### PR TITLE
Changing parse for '#' replace to match for both 'X' and 'x'

### PR DESCRIPTION
**Pull request by lucasbrendel**
This was found while testing other infineon XMC SVD files. From what i could tell, this should be allowed with the SVD schema, but correct me if this is allowing another edge case

### CODE REVIEW

File: src/parse.rs [hunk 1]
```diff
@@ -9,7 +9,12 @@ pub fn u32(tree: &Element) -> Option<u32> {
         // Handle strings in the binary form of:
         // #01101x1
         // along with don't care character x (replaced with 0)
-        u32::from_str_radix(&str::replace(&text["#".len()..], "x", "0"), 2).ok()
```
Comment (Emilgardis) [inline src/parse.rs]: I'd prefer

```rust
&text.to_lowercase()["#".len()..]
```

Comment (lucasbrendel) [reply src/parse.rs]: That is a much better approach. I can adjust that.

**Comment by japaric** [comment]
Thanks, @lucasbrendel. I'm not sure if this is actually allowed by the SVD spec but LGTM.

@homunkulus r+

**Comment by homunkulus** [comment]
:pushpin: Commit fd1ce6a has been approved by `japaric`

**Comment by homunkulus** [comment]
:hourglass: Testing commit fd1ce6a77d74b49988c5e8b075348945cf417211 with merge 53f25106e9260827039b518a17b8834979e8a682...

**Comment by homunkulus** [comment]
:sunny: Test successful - [status-travis](https://travis-ci.org/japaric/svd/builds/290174644?utm_source=github_status&utm_medium=notification)
Approved by: japaric
Pushing 53f25106e9260827039b518a17b8834979e8a682 to master...

**Pull request merged**